perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_freeipaGroupName.java

```diff
@@ -0,0 +1,89 @@
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.ResourceGroupAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.ResourceGroupAttributesModuleImplApi;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Module for attribute freeipaGroupName
+ *
+ * @author Peter Balcirak <peter.balcirak@gmail.com>
+ * @date 13.5.2016
+ */
+public class urn_perun_group_resource_attribute_def_def_freeipaGroupName extends ResourceGroupAttributesModuleAbstract implements ResourceGroupAttributesModuleImplApi {
+
+	private static final Pattern pattern = Pattern.compile("^[a-zA-Z0-9_.][a-zA-Z0-9_.-]{0,252}[a-zA-Z0-9_.$-]?$");
+	private static final String A_GR_freeipaGroupName = AttributesManager.NS_GROUP_RESOURCE_ATTR_DEF + ":freeipaGroupName";
+
+	@Override
+	public void checkAttributeValue(PerunSessionImpl sess, Resource resource, Group group, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+
+		//prepare group name and check its format
+		String groupName = (String) attribute.getValue();
+		if (groupName == null) {
+			throw new WrongAttributeValueException(attribute, group, "Attribute cannot be null.");
+		}
+
+		Matcher match = pattern.matcher(groupName);
+
+		if (!match.matches()) {
+			throw new WrongAttributeValueException(attribute, group, "Bad format of attribute freeipaGroupName. It has to match pattern ^[a-zA-Z0-9_.][a-zA-Z0-9_.-]{0,252}[a-zA-Z0-9_.$-]?$");
+		}
+
+		//Get facility for the resource
+		Facility facility = sess.getPerunBl().getResourcesManagerBl().getFacility(sess, resource);
+
+		// Get all resources from the facility
+		List<Resource> facilityResources =  sess.getPerunBl().getFacilitiesManagerBl().getAssignedResources(sess, facility);
+
+		//For each resource get all groups
+		for(Resource rs : facilityResources){
+			List<Group> resourceGroups =  sess.getPerunBl().getResourcesManagerBl().getAssignedGroups(sess, rs);
+
+			//For all groups get name and check uniqueness
+			for(Group gr : resourceGroups){
+				Attribute freeipaGroupNameAttribute = new Attribute();
+
+				try{
+					freeipaGroupNameAttribute = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, rs, gr, A_GR_freeipaGroupName);
+				} catch(AttributeNotExistsException ex) {
+					throw new ConsistencyErrorException("Attribute "+ A_GR_freeipaGroupName +" does not exists for group " + gr + " and resource " + rs ,ex);
+				}
+
+				if (freeipaGroupNameAttribute != null){
```

Comment:
This should never happen. Attribute will be not null there if exists. Only value can be null.